no-literal-checker/src/main/java/org/checkerframework/checker/noliteral/NoLiteralVisitor.java

```diff
@@ -0,0 +1,244 @@
+   * parameter lower bounds is top, which prevents the checker from defaulting exception parameters
+   * to @NonConstant. Overriding this method changes the expected lower bound to be @NonConstant,
+   * which prevents false positives.
+   *
+   * <p>Stolen shamelessly from https://github.com/awslabs/data-classification-checker
+   *
+   * @return a singleton set containing the @NonConstant annotation
+   */
+  @Override
+  protected Set<? extends AnnotationMirror> getExceptionParameterLowerBoundAnnotations() {
+    return Collections.singleton(atypeFactory.getNonConstant());
+  }
+
+  /**
+   * Overrides the default upper bound for thrown exceptions. By default, this is the same as the
+   * result of getExceptionParameterLowerBoundAnnotations, which defaults to top. This is overridden
+   * to keep the default top, since that other method is also overridden.
+   *
+   * <p>Stolen shamelessly from https://github.com/awslabs/data-classification-checker
+   *
+   * @return a singleton set containing the @MaybeDerivedFromConstant annotation
+   */
+  @Override
+  protected Set<? extends AnnotationMirror> getThrowUpperBoundAnnotations() {
+    return Collections.singleton(atypeFactory.getMaybeConstant());
+  }
+
+  /**
+   * All annotation arguments must be literals, so there is no need to check them. See
+   * https://github.com/typetools/checker-framework/issues/3178 for an explanation of why this is
+   * necessary to avoid false positives.
+   */
+  @Override
+  public Void visitAnnotation(AnnotationTree node, Void p) {
+    return null;
+  }
+
+  /**
+   * Searches through a type for user-written (i.e. non-default) annotations. Returns true if any
+   * are found.
+   *
+   * <p>Usage: NonDefaultScanner scanner = new NonDefaultScanner(); scanner.visit(type); boolean
+   * result = scanner.getResult();
+   */
+  private class NonDefaultScanner extends AnnotatedTypeScanner<Void, Void> {
+
+    private boolean result = false;
+
+    /**
+     * After calling visit(), this will return whether or not the visited type(s) have any non
+     * default annotations.
+     */
+    public boolean getResult() {
+      return result;
+    }
+
+    @Override
+    protected Void scan(AnnotatedTypeMirror type, Void aVoid) {
+      if (!type.hasAnnotation(NonConstant.class)) {
+        result = true;
+      }
+      return super.scan(type, aVoid);
+    }
+  }
+
+  /**
+   * Searches through a type for user-written (i.e. non-default) annotations. Returns true if any
+   * are found.
+   *
+   * @param atm the type to search
+   * @return true if the type or any of its components has a non-default annotation
+   */
+  private boolean hasNonDefault(final AnnotatedTypeMirror atm) {
+    NonDefaultScanner scanner = new NonDefaultScanner();
+    scanner.visit(atm);
+    return scanner.getResult();
+  }
+
+  /**
+   * This checker defaults implicit upper bounds of type variables to @NonConstant. This is the
```

Comment:
I can't find code that implements such a default. Either that's the standard default that the Checker Framework always uses, or this comment is wrong. I suspect the comment is wrong, because the example below refers to implicit upper bounds of type variables, which are a CLIMB-to-top location.

I removed this code and re-ran the tests, and got the same results. So I'm going to remove it.